pallets/amm/src/lib.rs

```diff
@@ -865,8 +868,20 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
             .ok_or(Error::<T, I>::InsufficientLiquidity)?;
 
         T::Assets::burn_from(pool.lp_token_id, who, liquidity)?;
-        T::Assets::transfer(base_asset, &Self::account_id(), who, base_amount, false)?;
-        T::Assets::transfer(quote_asset, &Self::account_id(), who, quote_amount, false)?;
+        T::Assets::transfer(
+            base_asset,
+            &Self::account_id(),
+            who,
+            base_amount,
+            base_asset == T::GetNativeCurrencyId::get(), // should keep alive if is native
+        )?;
+        T::Assets::transfer(
+            quote_asset,
+            &Self::account_id(),
```

Comment:
do we need to keep alive pallet account as well? in that case, there will be 0.01HKO in pool which cannot be swapped, but that's not an issue I guess

Reply:
yea I wasn't sure about that. my initial concern was that we could accidentally delete the pallet account from the chain while it still has assets. So I left the keep_alive check on since I don't think it will cause issues